docs: update subcommand plugin guide (#205)
sapphiredev/website: docs/Guide/plugins/Subcommands/subcommand-groups.mdx

---

> title: Using Subcommand Groups for Chat Input Commands

Reviewer: But message commands can use them too!

Author: Added a message command section.

sidebar_position: 2
---

> The following example will build off the command in the previous section. As a disclaimer, the result on the user side
> will seem redundant when compared to the previous command. Regardless, the purpose is to show how to use Subcommand
> Groups, rather than a useful example.

Reviewer: ```suggestion
The following examples shows off how you can implement support for mapping subcommands that are part of a subcommand group
to a method in your command class. This example does build off of the previous section, so if you jumped here
directly without checking it out, we suggest you go back there, take a peek, and come back fresh and ready to code!
```

> Since Subcommand Groups are a little nuanced, please refer to Discord's
> [Subcommands and Subcommand Groups](https://discord.com/developers/docs/interactions/application-commands#subcommands-and-subcommand-groups)
> documentation to see better examples of applying them.

Reviewer: ```suggestion
Subcommand groups are a way to...well...group subcommands by some criteria you want (for instance all subcommands affecting users can go into one `user` group). Discord's
[Subcommands and Subcommand Groups](https://discord.com/developers/docs/interactions/application-commands#subcommands-and-subcommand-groups) explanation might give you some better examples for how you can use them for your users' benefits! Here, we are just showing you how to implement the mappings for them using the Subcommands plugin! 
```

Author: I rewrote this when adding the message commands section, let me know if there are other changes needed.

## Chat Input Command Subcommand Groups Usage

:::caution

> To use subcommands with Chat Input Commands (Slash Commands), we recommend you learn how to register and use regular
> Chat Input Commands first in order to understand `registerApplicationCommands()`. Please read [Registering Chat Input
> Commands][register] for details.

Reviewer: ```suggestion
Before you can use subcommands with Chat Input Commands (Slash Commands), we recommend you learn how to register and use regular
Chat Input Commands first in order to understand what the `registerApplicationCommands()` method does. Please read [Registering Chat Input
Commands][register] for details, and once you're done, come back here!
```
